praw/models/reddit/redditor.py

```diff
@@ -76,12 +80,13 @@ class Redditor(
 
     STR_FIELD = "name"
 
-    @classmethod
-    def from_data(cls, reddit, data):
+    @staticmethod
+    def check_name(name: Union[str, Dict]):
```

Comment:
I don't think we should change the public interface of this class. That's a breaking change.

Reply:
I mean, the new descriptor is way better. Maybe keep both, but throw a DeprecationWarning similar to `mod_actions.py`.

Reply:
Even if the name is better, we can't change up the interface because that will break scripts that use PRAW. And for what it's worth, the name `from_data` is clear that it is creating an instance of the class from some data, while I don't know from the name what `check_name` does.

I guess I'm confused why you replaced `from_data` at all, because `from_data` and `check_name` seem to do two totally different things.

Reply:
I replaced from_data because it does not work with the `_create_base` private method in ExportableMixin, which requires that there be a keyword `_data` argument.

Reply:
Can you find a way to make it work? I really do not think that `from_data` should be removed or otherwise changed in a non-backwards-compatible way.

Reply:
Maybe add an optional _data parameter that does not need to be included.

Reply:
Regardless of how you do it, what I am saying is that `from_data` should still exist as a class method with the same (or possible extended, but not changed or shrunk) argument list and return type.

Reply:
I see. Adding the optional parameter should not affect old workflows.